### PR TITLE
Additional YAML serialization format tests

### PR DESCRIPTION
**Pull request by divergentdave**
This adds a few test cases to enforce backwards compatibility of YAML serialization of a few value-carrying enums. If we end up changing YAML libraries (see #2919), I figure these are the most likely sources of compatibility issues, both because use of tags is uncommon, and because we opt out of YAML tags in certain places with a "singleton map" adapter.

I also fixed a dev-dependency so that `cargo test -p janus_aggregator` works again.

### CODE REVIEW

File: aggregator/Cargo.toml [hunk 1]
```diff
@@ -115,6 +115,7 @@ janus_aggregator = { path = ".", features = ["fpvec_bounded_l2", "test-util"] }
 janus_aggregator_core = { workspace = true, features = ["test-util"] }
 mockito = { workspace = true }
 opentelemetry_sdk = { workspace = true, features = ["testing"] }
+prio = { workspace = true, features = ["multithreaded"] }
```
Comment (inahga) [inline aggregator/Cargo.toml:118]: This change seems unrelated?

**Review by inahga — APPROVED** [review]

File: aggregator/Cargo.toml (hunk 1 above)
Comment (inahga) [reply aggregator/Cargo.toml:118]: Oh oops I should read the PR description.

**Review by branlwyd — APPROVED** [review]
